### ISSUE TITLE
Error when training range is too small

### ISSUE THREAD

**Issue by daradermody** [regel/loudml#72]
When a small enough range is used, the following error occurs:
```
INFO:hyperopt.tpe:tpe_transform took 0.002254 seconds
INFO:hyperopt.tpe:TPE using 49/49 trials with best loss inf
WARNING:root:iteration failed: Error when checking input: expected lstm_1_input to have 3 dimensions, but got array with shape (0, 1)
ERROR:root:attempt to get argmin of an empty sequence
Traceback (most recent call last):
  File "/usr/lib/loudml/vendor/numpy/core/fromnumeric.py", line 51, in _wrapfunc
    return getattr(obj, method)(*args, **kwds)
AttributeError: 'list' object has no attribute 'argmin'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/lib/python3.6/dist-packages/loudml/worker.py", line 53, in run
  File "/usr/lib/python3.6/dist-packages/loudml/worker.py", line 89, in train
  File "/usr/lib/python3.6/dist-packages/loudml/timeseries.py", line 955, in train
  File "/usr/lib/python3.6/dist-packages/loudml/timeseries.py", line 807, in _train_on_dataset
  File "/usr/lib/loudml/vendor/hyperopt/fmin.py", line 307, in fmin
    return_argmin=return_argmin,
  File "/usr/lib/loudml/vendor/hyperopt/base.py", line 635, in fmin
    return_argmin=return_argmin)
  File "/usr/lib/loudml/vendor/hyperopt/fmin.py", line 322, in fmin
    return trials.argmin
  File "/usr/lib/loudml/vendor/hyperopt/base.py", line 590, in argmin
    best_trial = self.best_trial
  File "/usr/lib/loudml/vendor/hyperopt/base.py", line 585, in best_trial
    best = np.argmin(losses)
  File "/usr/lib/loudml/vendor/numpy/core/fromnumeric.py", line 1101, in argmin
    return _wrapfunc(a, 'argmin', axis=axis, out=out)
  File "/usr/lib/loudml/vendor/numpy/core/fromnumeric.py", line 61, in _wrapfunc
    return _wrapit(obj, method, *args, **kwds)
  File "/usr/lib/loudml/vendor/numpy/core/fromnumeric.py", line 41, in _wrapit
    result = getattr(asarray(obj), method)(*args, **kwds)
ValueError: attempt to get argmin of an empty sequence
127.0.0.1 - - [2018-11-07 10:45:32] "POST /models/ml-month/_train?from=2009-04-15T09%3A27%3A16.363Z&to=2015-03-09T04%3A21%3A49.090Z HTTP/1.1" 200 147 0.003137
ERROR:root:job[7a96561f-d8bb-40a2-9ad8-ecc2ea9bc28c] failed: attempt to get argmin of an empty sequence
```

I get the above when I create a model with less than around 100 buckets.

Should there be a more appropriate error (like "not enough data") and is there any way to determine how much data is needed by Loud ML for some metric (maybe it's a specific minimum number of agg buckets)?

**Comment by regel** [comment]
Hi @daradermody, you read my mind. Agreed, "not enough data" message would provide a better UX in this scenario. Let's target 1.4.4 to fix this.

**Comment by regel** [comment]
fixed in 1.5.0, a specific exception was added in the training code to raise a more informative message.
